### PR TITLE
feat(releases, core): supporting remotely deleted bundles

### PR DESCRIPTION
**Pull request by jordanl17**
### Description
When a bundle is deleted by another user, and your user has not yet refreshed the studio:
In the perspective global selector, the deleted bundle will show:
Screenshot 2024-08-08 at 16 44 31
In the Release Overview list in Releases Tool, the bundle will show:
Screenshot 2024-08-08 at 16 44 37
If you are on the Document form for a document which had been in the deleted bundle, the toast will show:
Screenshot 2024-08-08 at 17 33 03

### What to review
* Update to the bundle store state to persist deleted bundles (if they already exist locally and have been deleted by another user)

### Testing
Relevant tests have been updated and new cases added for deleted bundles

### Notes for release
N/A

### CODE REVIEW

File: packages/sanity/src/structure/panes/document/document-layout/useBundleDeleted.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import {Text, useToast} from '@sanity/ui'
+import {useEffect, useState} from 'react'
+import {Translate, useBundles, usePerspective, useTranslation} from 'sanity'
+
+export const useBundleDeleted = () => {
```
Comment (pedrobonamin) [inline packages/sanity/src/structure/panes/document/document-layout/useBundleDeleted.tsx:5]: I think renaming this to `useBundleDeletedToast` would make it easier to quickly understand what this hook is doing.

**Review by pedrobonamin — COMMENTED** [review]
Thanks for this @jordanl17 it's looking very good, aded some suggestions

**Review by jordanl17 — COMMENTED** [review]
@pedrobonamin thanks for the review. I've:
* renamed to `useBundleDeletedToast` (as agreed will move this to a banner after this is merged)
* Removed the unnecessary state in `useBundleDeletedToast`
* Added an id and allowed for closable on these toast
* Added in a disabled test to `BundleMenu`
* Avoid showing deleted bundles in `DocumentPerspectiveMenu`

File: packages/sanity/src/core/bundles/components/BundleMenu.tsx [hunk 2]
```diff
@@ -30,11 +33,28 @@ interface BundleListProps {
  */
 export function BundleMenu(props: BundleListProps): JSX.Element {
   const {bundles, loading, actions, button} = props
+  const {deletedBundles} = useBundles()
```
[the remark this answers is not in the document]
Comment (pedrobonamin) [reply packages/sanity/src/core/bundles/components/BundleMenu.tsx:36]: sounds reasonable! Thanks!

**Comment by pedrobonamin** [comment]
> @pedrobonamin thanks for the review. I've:
> 
> * renamed to `useBundleDeletedToast` (as agreed will move this to a banner after this is merged)
> * Removed the unnecessary state in `useBundleDeletedToast`
> * Added an id and allowed for closable on these toast
> * Added in a disabled test to `BundleMenu`
> * Avoid showing deleted bundles in `DocumentPerspectiveMenu`

Thanks @jordanl17 , it looks good to me, let's add a follow up ticket for the banner.

**Review by pedrobonamin — APPROVED** [review]

**Pull request merged**